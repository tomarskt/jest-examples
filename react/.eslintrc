{
  "env": {
    "node": true,
    "es6": true,
    "jest/globals": true
  },
  "extends": [
    "eslint:recommended",
    "airbnb-base"
  ],
  "plugins": [
    "jest"
  ],
  "rules": {
    "quotes": "off",
    "no-console": "off",
    "no-debugger": "off",
    "indent": ["off",4 ]
  }
}