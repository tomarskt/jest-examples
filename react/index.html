<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charSet="utf-8"/>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <title>issue tracker</title>
    <link href="/style.css" rel='stylesheet' />
  </head>
  <body>
    <div id="root" />
    <script type="text/javascript" src="/bundle.js"></script>
  </body>
</html>