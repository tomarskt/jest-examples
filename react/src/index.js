import React, {Component} from 'react';
import ReactDom from 'react-dom';
import './styles/index.less'

import IssueManager from './components/issueManager';
class App extends Component {
    render() {
        return <IssueManager />
    }
}

ReactDom.render(<App/>,document.querySelector('#root'));