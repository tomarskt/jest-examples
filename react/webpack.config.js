var path = require('path');
var webpack = require('webpack');
const autoprefixer = require('autoprefixer')
const ExtractTextPlugin = require('extract-text-webpack-plugin')

var config = {
  devtool: 'source-map',
  devServer: {
    historyApiFallback: true
  },
  entry: './src/index.js',
  output: {
    path: path.resolve('./dist'),
    filename: 'bundle.js',
    publicPath: '/'
  },
  plugins: [
    new webpack.HotModuleReplacementPlugin(),
    new ExtractTextPlugin('style.css')
  ],
  module: {
    rules: [ // "style-loader!css-loader!less-loader" can only be used with module.loaders;
      {
        test: /\.jsx?/,
        use: ['babel-loader'],
        exclude: /node_modules/,
      },
      {
        test: /\.less$/,
        use: ExtractTextPlugin.extract({
          fallback: "style-loader",
          use: ["css-loader",
            {
              loader: 'postcss-loader',
              options: {
                plugins:[
                  autoprefixer({
                    browsers: ['last 2 versions']
                  })
                ]
              }
            },
            "less-loader"]
        })
      }
    ]
  },
};


module.exports = config