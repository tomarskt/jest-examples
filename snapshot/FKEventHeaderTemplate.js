import React from 'react'
import RaceTimeStatus from 'Frankel/components/RaceTimeStatus'

export function FKEventHeaderTemplate(props) {

    return (<div>
        <RaceTimeStatus event={props.event} />      
    </div>)
}