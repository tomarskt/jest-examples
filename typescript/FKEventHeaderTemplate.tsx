import React from 'react'; // for typescript to work, module must have module.exports.default={xyz}
import RaceTimeStatus from 'Frankel/components/RaceTimeStatus'

export function FKEventHeaderTemplate(props) {
    console.log(RaceTimeStatus)
    return (<div>
       <RaceTimeStatus event={props.event}/>
    </div>)
}