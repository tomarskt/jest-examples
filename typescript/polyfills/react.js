var react = require('react');
// short for module.exports;
exports = react; // targeting CommonJS consumers; can do require('./react');
module.exports.default=react;  // targeting TypeScript consumers; can do import React from './react';