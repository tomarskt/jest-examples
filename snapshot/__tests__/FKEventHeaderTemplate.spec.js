import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {FKEventHeaderTemplate} from '../FKEventHeaderTemplate'

// you must specify {virtual:true} becuase file 'Frankel/components/RaceTimeStatus' doesn't exist
// error :Cannot find module 'Frankel/components/RaceTimeStatus' from 'FKEventHeaderTemplate.spec.js'
jest.mock('Frankel/components/RaceTimeStatus', () => 'RaceTimeStatus', {virtual: true});

describe('FKEventHeaderTemplate Component', () => {
    it('should render component correctly',() => {
        const props={
            event:{
                id:'123'
            }
        }
        const component = renderer.create( <FKEventHeaderTemplate  {...props}/> );
        expect(component.toJSON()).toMatchSnapshot()
    });
});
