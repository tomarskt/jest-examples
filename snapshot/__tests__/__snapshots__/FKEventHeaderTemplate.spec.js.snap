// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FKEventHeaderTemplate Component should render component correctly 1`] = `
<div>
  <RaceTimeStatus
    event={
      Object {
        "id": "123",
      }
    }
  />
</div>
`;
